crates/autopilot/src/decoded_settlement.rs

```diff
@@ -0,0 +1,511 @@
+//! This module contains the logic for decoding the function input for
+//! GPv2Settlement::settle function.
+
+use {
+    anyhow::{anyhow, Context, Result},
+    bigdecimal::{Signed, Zero},
+    contracts::GPv2Settlement,
+    ethcontract::{common::FunctionExt, tokens::Tokenize, Address, Bytes, U256},
+    model::order::{Order, OrderKind},
```

Comment:
Since this is new code I'd like to consider not using the `model::Order` struct. That type contains a lot of information, most of which is not relevant for this component. We should treat it as the order type for the API. For a component like this we can define a new Order struct with the minimal information it needs to work. It can implement `From<model::Order>`. This will make this component easier to maintain.

Reply:
Yes, makes sense.